stablehlo/dialect/StablehloOps.td

```diff
@@ -1282,7 +1282,8 @@ def StableHLO_WhileOp: StableHLO_Op<"while", [
 }
 
 def StableHLO_AllGatherOp : StableHLO_Op<"all_gather",
-    [SameOperandsAndResultElementType] /*all_gather_c6*/> {
+    [SameOperandsAndResultElementType,
+     HLO_MatchingOperandsAndResultScaleZeroPoint] /*all_gather_c6*/> {
```

Comment:
We do not need this here? `SameOperandsAndResultElementType` should take care of equal scales and zero-points. Please confirm before removing.

Reply:
Can we just use the existing `SameOperandsAndResultElementType` everywhere else? Does this trait only expects one result or something?

Reply:
@GleasonK That is a good point!

It seems that in most of the case we do not need this new trait and can use  the existing one `SameOperandsAndResultElementType`. This is true for all the operations with constraints similar to `type(operand...) = type(result..)`. 

@abhigunj 

How about we:
1. Try reusing the existing traits instead of the new one. Apply the change on all operations which need same operands and result quantized type.
3. If no surprises on (1), update the issue label and description.

Reply:
Thank you all for the suggestions. 
took a relook at ops with type(operand...) = type(result..) constraints which need verifier updates: 

* `SameOperandsAndResultElementType` Trait is used for the three OPs (`collective_broadcast`, `collective_permute`, `reverse`). Validated.  Will use this PR to update it. 

* `reshape` `transpose` `broadcast_in_dim`   we can't use  `SameOperandsAndResultElementType`  due to slight variation from the `type(operand..)` constraints. I have updated their custom verifier at https://github.com/openxla/stablehlo/pull/2081 . Plan was to replace those with the new Trait. We will keep updates to custom verifier unchanged  and won't add new Trait from this PR.